### PR TITLE
Basic swerve

### PR DESCRIPTION
**Pull request by linglejack06**
- add PhoenixDriveConstants file to handle configuration
- add basic Phoenix Drive Subsystem: no alignment, auto, or special Commands
-

Resolves Issue #2

### CODE REVIEW

File: src/main/java/frc/robot/BuildConstants.java
Comment (jkleiber) [inline src/main/java/frc/robot/BuildConstants.java:1]: can we gitignore this

File: src/main/java/frc/robot/subsystems/PhoenixDrive.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            DriverStation.getAlliance()
+                    .ifPresent(
+                            (color) -> {
+                                this.setOperatorPerspectiveForward(
+                                        color == Alliance.Red
+                                                ? RedAlliancePerspectiveRotation
+                                                : BlueAlliancePerspectiveRotation);
+                            });
+            hasAppliedOperatorPerspective = true;
+        }
```
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/PhoenixDrive.java:75]: what is this?

**Review by jkleiber — COMMENTED** [review]
a couple comments - also does this work in sim?

File: src/main/java/frc/robot/subsystems/PhoenixDrive.java (hunk 1 above)
Comment (linglejack06) [reply src/main/java/frc/robot/subsystems/PhoenixDrive.java:75]: this was in Phoenix 6's example, I was trying to figure it out myself. I think its for field relative positioning within the SwerveDrivetrain class.

**Comment by linglejack06** [comment]
@jkleiber this request has no sim. However, I already have the sim on another branch built. Haven't tested yet though. I figured I would push that basic stuff to main then get the sim and auto done under separate pull request

File: src/main/java/frc/robot/subsystems/PhoenixDrive.java (hunk 1 above)
Comment (jkleiber) [reply src/main/java/frc/robot/subsystems/PhoenixDrive.java:75]: okay, I guess we'll see what happens when we test this in sim anyways

**Comment by linglejack06** [comment]
@jkleiber will mentioned something about this pull request but he couldn't remember what it was. He had said you asked me to do something so we could get it merged? Is it just the comments?

**Comment by jkleiber** [comment]
@linglejack06 can we just gitignore and untrack the autogenerated constants file (then this will be good to merge imo)

**Comment by linglejack06** [comment]
@jkleiber it should be fixed now

**Review by jkleiber — APPROVED** [review]

**Pull request merged**